### DISCUSSION TITLE
Access all xml data of mujoco xml files from pybullet

### DISCUSSION THREAD

**Discussion by benelot** [bulletphysics/bullet3#3759] [General]
As you may know, I am working on a approximately compatible port of the mujoco envs of openai gym to replace mujoco as the prevalent physics engine. I am planning to rewrite my ports in pybulletgym to allow direct usage of the envs implemented by openai. This would for example mean taking inverse pendulum env (https://github.com/openai/gym/blob/master/gym/envs/mujoco/inverted_pendulum.py) and just replace the MujocoEnv it inherits from with the BulletEnv which implements the same methods required for all other envs.

For this to work properly, I wanted to ask if pybullet supports some way to read all values from the mujoco xml files. Since pybullet already parses the file, I think it would be easy to just return the not used keys and values as anpython dictionnary. Doing this could allow me to read and simulate properties set in mujoco. Is there a way to access the xml structure or its values? Would you be interested in an implementation of something like it?

**Comment by erwincoumans** [comment]
It is a good point, and I'd like to see that feature too. Someone is looking into adding user values, and I'll discuss this request. Thanks for the report.
